cpp-ch/local-engine/Parser/SerializedPlanParser.cpp

```diff
@@ -160,20 +160,31 @@ void SerializedPlanParser::adjustOutput(const DB::QueryPlanPtr & query_plan, con
             else
             {
                 need_final_project = true;
-
                 bool need_const = origin_column.column && isColumnConst(*origin_column.column);
-                ColumnWithTypeAndName final_column(
-                    need_const ? final_type->createColumnConst(0, assert_cast<const ColumnConst &>(*origin_column.column).getField())
-                               : final_type->createColumn(),
-                    final_type,
-                    origin_column.name);
-                final_columns.emplace_back(std::move(final_column));
+                if (need_const)
+                {
+                    const DB::ContextPtr context = DB::CurrentThread::get().getQueryContext();
+                    const FunctionOverloadResolverPtr & cast_resolver = FunctionFactory::instance().get("CAST", context);
+                    const DataTypePtr string_type = DB::DataTypeFactory::instance().get("String");
+                    ColumnWithTypeAndName to_type_column = {string_type->createColumnConst(1, final_type->getName()), string_type, "__cast_const__"};
+                    FunctionBasePtr cast_function = cast_resolver->build({origin_column, to_type_column});
+                    ColumnPtr cast_col = cast_function->execute({origin_column, to_type_column}, final_type, 1);
+                    Field cast_field;
+                    cast_col->get(0, cast_field);
+                    ColumnWithTypeAndName final_column(final_type->createColumnConst(0, cast_field), final_type, origin_column.name);
```

Comment:
maybe we can wrap `ColumnConst` around `cast_col` directly without extracting field from it.

Reply:
thanks, it's more concise and clear